### ISSUE TITLE
SUMA: Intercepted X11 error: BadWindow (invalid Window parameter)

### ISSUE THREAD

**Issue by yarikoptic** [afni/afni#131]
cmake build, as of AFNI_19.3.12-13-g8673efbd6 (currently `fix_suma_and_add_system_install_management` branch of https://github.com/leej3/afni
), suma fails to start with

```
(git)lena:…ppsy/afni-upstream[fix_suma_and_add_system_install_management]obj-build
$> src/suma                 

suma: 
     No input specified, loading some toy surfaces...
     Use '.' and ',' to cycle between them.
     See suma -help for assistance.

oo     Warning SUMA_XErrHandler (../src/SUMA/SUMA_display.c:4148):
Intercepted X11 error: BadWindow (invalid Window parameter)
Will attempt to proceed but trouble might ensue.
++     Notice SUMA_XErrHandler (../src/SUMA/SUMA_display.c:4150 @11:32:49):
Trace At Xerr
    SUMA_XErrHandler
   SUMA_X_SurfaceViewer_Create
  suma
 Bottom of Debug Stack
oo     Warning SUMA_XErrHandler (../src/SUMA/SUMA_display.c:4148):
Intercepted X11 error: BadWindow (invalid Window parameter)
Will attempt to proceed but trouble might ensue.
++     Notice SUMA_XErrHandler (../src/SUMA/SUMA_display.c:4150 @11:32:49):
Trace At Xerr
    SUMA_XErrHandler
   SUMA_X_SurfaceViewer_Create
  suma
 Bottom of Debug Stack
oo     Warning SUMA_XErrHandler (../src/SUMA/SUMA_display.c:4148):
Intercepted X11 error: BadWindow (invalid Window parameter)
Will attempt to proceed but trouble might ensue.
++     Notice SUMA_XErrHandler (../src/SUMA/SUMA_display.c:4150 @11:32:49):
Trace At Xerr
    SUMA_XErrHandler
   SUMA_X_SurfaceViewer_Create
  suma
 Bottom of Debug Stack
Error: No realize class procedure defined
```

it was built on debian testing/sid with
```
mkdir -p obj-build && cd obj-build && cmake -GNinja -DBUILD_BINARIES=ON -DBUILD_X_DEPENDENT_GUI_PROGS=ON -DBUILD_OPENGL_DEPENDENT_GUI_PROGS=ON -DBUILD_PLUGINS=ON -DUSE_OMP=ON -DUSE_SYSTEM_XMHTML=ON .. && time ninja && ninja pytest
```
happen you decide to reproduce

Any ideas on how to troubleshoot it?  Just in case:

<details>
<summary>ldd on suma</summary> 

```shell
$> ldd src/suma                                                                                                                                                                                                
	linux-vdso.so.1 (0x00007ffeabf6d000)
	libmrix.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libmrix.so (0x00007f16f9c45000)
	libmri.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libmri.so (0x00007f16f8b80000)
	libX11.so.6 => /lib/x86_64-linux-gnu/libX11.so.6 (0x00007f16f8a0d000)
	libXt.so.6 => /lib/x86_64-linux-gnu/libXt.so.6 (0x00007f16f89a2000)
	libglut.so.3 => /lib/x86_64-linux-gnu/libglut.so.3 (0x00007f16f8955000)
	libGLw.so.1 => /lib/x86_64-linux-gnu/libGLw.so.1 (0x00007f16f874e000)
	libSUMA.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libSUMA.so (0x00007f16f78f0000)
	libOpenGL.so.0 => /lib/x86_64-linux-gnu/libOpenGL.so.0 (0x00007f16f78bf000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f16f76ff000)
	libcoxplot.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libcoxplot.so (0x00007f16f75b5000)
	libXmHTML.so.1.1 => /lib/x86_64-linux-gnu/libXmHTML.so.1.1 (0x00007f16f7335000)
	libXm.so.4 => /lib/x86_64-linux-gnu/libXm.so.4 (0x00007f16f6e67000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f16f6d20000)
	libnetcdf.so.13 => /lib/x86_64-linux-gnu/libnetcdf.so.13 (0x00007f16f6bdf000)
	lib3DEdge.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/lib3DEdge.so (0x00007f16f6bc4000)
	libeispack.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libeispack.so (0x00007f16f6b7e000)
	libvolpack.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libvolpack.so (0x00007f16f6b1c000)
	libnifti2.so.2.8.9 => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libnifti2.so.2.8.9 (0x00007f16f6aea000)
	libf2cdir.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libf2cdir.so (0x00007f16f6ac8000)
	libnifticdf.so.2.8.9 => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libnifticdf.so.2.8.9 (0x00007f16f6aa2000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f16f6a85000)
	libxcb.so.1 => /lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f16f6a5b000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f16f6a56000)
	libSM.so.6 => /lib/x86_64-linux-gnu/libSM.so.6 (0x00007f16f6a49000)
	libICE.so.6 => /lib/x86_64-linux-gnu/libICE.so.6 (0x00007f16f682c000)
	libGL.so.1 => /lib/x86_64-linux-gnu/libGL.so.1 (0x00007f16f6798000)
	libXext.so.6 => /lib/x86_64-linux-gnu/libXext.so.6 (0x00007f16f6586000)
	libXi.so.6 => /lib/x86_64-linux-gnu/libXi.so.6 (0x00007f16f6376000)
	libXxf86vm.so.1 => /lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007f16f6170000)
	libtrack_tools.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libtrack_tools.so (0x00007f16f6054000)
	libsegtools.so => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libsegtools.so (0x00007f16f6033000)
	libGLU.so.1 => /lib/x86_64-linux-gnu/libGLU.so.1 (0x00007f16f5fc2000)
	libXmu.so.6 => /lib/x86_64-linux-gnu/libXmu.so.6 (0x00007f16f5fa5000)
	libGLX.so.0 => /lib/x86_64-linux-gnu/libGLX.so.0 (0x00007f16f5f71000)
	libGLdispatch.so.0 => /lib/x86_64-linux-gnu/libGLdispatch.so.0 (0x00007f16f5eb4000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f16f9de6000)
	libXpm.so.4 => /lib/x86_64-linux-gnu/libXpm.so.4 (0x00007f16f5ca0000)
	libXft.so.2 => /lib/x86_64-linux-gnu/libXft.so.2 (0x00007f16f5a8a000)
	libfontconfig.so.1 => /lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007f16f5a44000)
	libjpeg.so.62 => /lib/x86_64-linux-gnu/libjpeg.so.62 (0x00007f16f57db000)
	libpng16.so.16 => /lib/x86_64-linux-gnu/libpng16.so.16 (0x00007f16f57a2000)
	libhdf5_serial_hl.so.100 => /lib/x86_64-linux-gnu/libhdf5_serial_hl.so.100 (0x00007f16f577a000)
	libhdf5_serial.so.103 => /lib/x86_64-linux-gnu/libhdf5_serial.so.103 (0x00007f16f53f5000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f16f53d4000)
	libcurl-gnutls.so.4 => /lib/x86_64-linux-gnu/libcurl-gnutls.so.4 (0x00007f16f5346000)
	libznz.so.2.8.9 => /home/yoh/deb/gits/pkg-exppsy/afni-upstream/obj-build/src/libznz.so.2.8.9 (0x00007f16f5341000)
	libXau.so.6 => /lib/x86_64-linux-gnu/libXau.so.6 (0x00007f16f513b000)
	libXdmcp.so.6 => /lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f16f4f35000)
	libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007f16f4f2c000)
	libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007f16f4f12000)
	libgsl.so.23 => /lib/x86_64-linux-gnu/libgsl.so.23 (0x00007f16f4c89000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f16f4aae000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f16f4a94000)
	libfreetype.so.6 => /lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007f16f49d5000)
	libXrender.so.1 => /lib/x86_64-linux-gnu/libXrender.so.1 (0x00007f16f47cb000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f16f479e000)
	libsz.so.2 => /lib/x86_64-linux-gnu/libsz.so.2 (0x00007f16f4797000)
	libnghttp2.so.14 => /lib/x86_64-linux-gnu/libnghttp2.so.14 (0x00007f16f476f000)
	libidn2.so.0 => /lib/x86_64-linux-gnu/libidn2.so.0 (0x00007f16f474f000)
	librtmp.so.1 => /lib/x86_64-linux-gnu/librtmp.so.1 (0x00007f16f4730000)
	libssh2.so.1 => /lib/x86_64-linux-gnu/libssh2.so.1 (0x00007f16f4702000)
	libpsl.so.5 => /lib/x86_64-linux-gnu/libpsl.so.5 (0x00007f16f46ef000)
	libnettle.so.7 => /lib/x86_64-linux-gnu/libnettle.so.7 (0x00007f16f46b2000)
	libgnutls.so.30 => /lib/x86_64-linux-gnu/libgnutls.so.30 (0x00007f16f44ed000)
	libgssapi_krb5.so.2 => /lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007f16f44a1000)
	libldap_r-2.4.so.2 => /lib/x86_64-linux-gnu/libldap_r-2.4.so.2 (0x00007f16f444c000)
	liblber-2.4.so.2 => /lib/x86_64-linux-gnu/liblber-2.4.so.2 (0x00007f16f443b000)
	libbrotlidec.so.1 => /lib/x86_64-linux-gnu/libbrotlidec.so.1 (0x00007f16f442d000)
	libgslcblas.so.0 => /lib/x86_64-linux-gnu/libgslcblas.so.0 (0x00007f16f43ea000)
	libaec.so.0 => /lib/x86_64-linux-gnu/libaec.so.0 (0x00007f16f43e1000)
	libunistring.so.2 => /lib/x86_64-linux-gnu/libunistring.so.2 (0x00007f16f425f000)
	libhogweed.so.5 => /lib/x86_64-linux-gnu/libhogweed.so.5 (0x00007f16f4226000)
	libgmp.so.10 => /lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f16f41a3000)
	libgcrypt.so.20 => /lib/x86_64-linux-gnu/libgcrypt.so.20 (0x00007f16f4084000)
	libp11-kit.so.0 => /lib/x86_64-linux-gnu/libp11-kit.so.0 (0x00007f16f3f54000)
	libtasn1.so.6 => /lib/x86_64-linux-gnu/libtasn1.so.6 (0x00007f16f3f3e000)
	libkrb5.so.3 => /lib/x86_64-linux-gnu/libkrb5.so.3 (0x00007f16f3e5e000)
	libk5crypto.so.3 => /lib/x86_64-linux-gnu/libk5crypto.so.3 (0x00007f16f3e2d000)
	libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2 (0x00007f16f3e27000)
	libkrb5support.so.0 => /lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007f16f3e16000)
	libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1 (0x00007f16f3e0f000)
	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007f16f3df7000)
	libsasl2.so.2 => /lib/x86_64-linux-gnu/libsasl2.so.2 (0x00007f16f3dda000)
	libbrotlicommon.so.1 => /lib/x86_64-linux-gnu/libbrotlicommon.so.1 (0x00007f16f3db7000)
	libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0 (0x00007f16f3d92000)
	libffi.so.6 => /lib/x86_64-linux-gnu/libffi.so.6 (0x00007f16f3d88000)

```
</details>

**Comment by leej3** [comment]
@yarikoptic, following your help demonstrating that it was an issue with link order (by using `LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libXm.so suma`). Unlike [before](https://github.com/afni/afni/pull/126/commits/a4955d12ae137853843fda6df885a3144215072b) the simple fix didn't work. Instead it looks like @mih 's [fix](https://github.com/neurodebian/afni/blob/79012f49ea4027b839750662c65483741344ac7a/debian/patches/cmake_sharedlibs-cmake#L899) for this is `-Wl,--no-as-needed ${MOTIF_LIBRARIES} -Wl,--as-needed` and is required over just specifying the link order correctly in cmake (and gcc).  I may write a Cmake find module for motif so that I can import the libraries as a target and have a this set as a link_option of the target to avoid having this happen again. From what I can tell this solves this issue without directly causing any other issue!

**Comment by leej3** [comment]
Solved in #137
